tools/hardware/config.py

```diff
@@ -48,7 +48,10 @@ def align_memory(self, regions: Set[Region]) -> List[Region]:
 class ARMConfig(Config):
     ''' Config class for ARM '''
     arch = 'arm'
-    SUPERSECTION_BITS = 24  # 2^24 = 16 MiByte
+
+    def __init__(self, sel4arch, addrspace_max):
+        super().__init__(sel4arch, addrspace_max)
+        self.SUPERSECTION_BITS = 25 if sel4arch == 'arm_hyp' else 24
 
     def get_kernel_phys_align(self) -> int:
         ''' on ARM the ELF loader expects to be able to map a supersection page to load the kernel. '''
```

Comment:
This comment doesn't seem right to me.
- On arm (non-hyp) the elfloader maps the kernel with 1Mib pages,
- On arm-hyp 2MiB pages are used,
- On aarch64 1GiB pages are used.

There's a static assert in the kernel that talks about supersections:
```

/* For alignment conditions to translate over addrFromPPtr, physBase must be
   aligned to at least the size of a SuperSection. */
compile_assert(physBase_aligned, IS_ALIGNED(PHYS_BASE_RAW, seL4_SuperSectionBits));
```

Is there a requirement that the kernel mapping window needs to at least preserve alignment of the largest page size for the architecture? (Assuming that that's the largest kernel object that has hardware alignment requirements).

Reply:
Yes, for AArch32, there is arithmetic that will fail (in the kernel) if the mapping window is not aligned to the largest page size. I'm not sure any more why that is not the case for AArch64, but it could be that the alignment there is hard coded to be much bigger anyway.

The comment in the file is old, I don't know where it's from, but I guess the alignment of the mapping window doesn't really have to coincide with the page size that is used.

Reply:
Ok, so we do know that the comment is wrong, because the ELF loader does not actually do what the comment says. I'll update it to say that it is a kernel requirement for AArch32.

Reply:
Now done.